apis/Google.Cloud.AspNetCore.DataProtection.SecretManager/Google.Cloud.AspNetCore.DataProtection.SecretManager/CloudSecretManagerXmlRepository.cs

```diff
@@ -46,75 +44,56 @@ internal CloudSecretManagerXmlRepository(SecretManagerServiceClient client, stri
         GaxPreconditions.CheckNotNull(projectId, nameof(projectId));
         _projectName = new ProjectName(projectId);
         _secretName = new SecretName(projectId, secretName);
-
-        // Version is populated in such a way that it always refers to the most latest version.
-        // For more information on version naming refer to
-        // https://cloud.google.com/dotnet/docs/reference/Google.Cloud.SecretManager.V1/latest/Google.Cloud.SecretManager.V1.AccessSecretVersionRequest#Google_Cloud_SecretManager_V1_AccessSecretVersionRequest_Name
-        _version = SecretVersionName.FromProjectSecretSecretVersion(projectId, secretName, "latest");
     }
 
     /// <inheritdoc />
     public IReadOnlyCollection<XElement> GetAllElements()
     {
-        if (!SecretExists())
+        try
         {
-            return Array.Empty<XElement>();
+            var elements = new List<XElement>();
+            foreach (SecretVersion secretVersion in _secretManagerServiceClient.ListSecretVersions(_secretName))
```

Comment:
I don't know enough about SecretManager to know whether this is a sensible way of achieving the aim of "storing a new element adds a new one". I'd have *expected* SecretManager to be built on "each version is intended to be complete", but I could be wrong. If I'm right though, I'd suggest one of two options:

- Create a new secret per element, using the secret name as a prefix
- Fetch the existing secret, parse it as an XDocument, add the element, and store the result (with suitable race condition handling)

I'd suggest talking to the SecretManager team about this.

Reply:
Thanks Jon for sharing your input. 
I am referring to the existing FileSystemXmlRespository code  and  trying to make our implementation look similar to that. In the same we can see that they are basically adding every XElement in a separate file 
ref: https://docs.google.com/document/d/1iVHN6VuF0pTDDoai1YUSgyVJ6rxrqL6yCMdwfk66740/edit#heading=h.bbjct62hz474

and while reading they are reading all the files with the extension .xml inside the folder.
ref: https://github.com/dotnet/aspnetcore/blob/main/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs#L84
In our implementation we can think of replacing directory with Secret and the file with the secret version. So every secret version will contain the XElement and while returning we will just return the list of versions available.

let me know your thoughts on the same.

Reply:
My point is that I don't think those are comparable concepts. There's a difference between "a directory with multiple independent files" and "multiple versions of one resource". Yes, you *can* list secret versions and get independent data, but I'm not convinced that's how SecretManager expects versions to be used. (Would you use separate commits in git to store fundamentally distinct data in the same file, for example? That would seem odd to me.)

Again, I'd suggest reaching out to the SecretManager team to discuss how multiple independent items should be stored in SecretManager.

Reply:
Ok Jon I got the point thanks for explanation Will reach out to the secret manager team as next step.